# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/base/content/browser.js
# Commit: 1cc86398ced0
# Full Hash: 1cc86398ced034fafc66d94799f91221420e1229
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2021-02-18 04:16:06
# Regressor Bug: 1692350
# File Overlap Count: 1
# Description:
#   Bug 1692350 - Implement a telemetry probe to collect page load data for documents that use lazyload r=emilio,chutten
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D104912
# ==============================================================================

diff -r 1ca4c3365fae -r 1cc86398ced0 browser/base/content/browser.js
--- a/browser/base/content/browser.js	Wed Feb 17 22:02:17 2021 +0000
+++ b/browser/base/content/browser.js	Wed Feb 17 22:25:02 2021 +0000
@@ -5776,6 +5776,15 @@
           stopwatchRunning /* we won't see STATE_START events for pre-rendered tabs */
         ) {
           if (recordLoadTelemetry) {
+            if (aBrowser.browsingContext?.topWindowContext?.hadLazyLoadImage) {
+              let timeElapsed = TelemetryStopwatch.timeElapsed(
+                histogram,
+                aBrowser
+              );
+              Services.telemetry
+                .getHistogramById("FX_LAZYLOAD_IMAGE_PAGE_LOAD_MS")
+                .add(timeElapsed);
+            }
             TelemetryStopwatch.finish(histogram, aBrowser);
             BrowserTelemetryUtils.recordSiteOriginTelemetry(browserWindows());
           }